### DISCUSSION TITLE
Cut video and preserve gyro data

### DISCUSSION THREAD

**Discussion by blurred82** [mifi/lossless-cut#1596] [Ideas]
Hi,
I would like to know is it possible to cut mp4 from insta360 Go2 camera, which records gyro data
to allow stabilize video in post. (Gyroflow, insta360 Studio...).
Unfortunetly cut file loses gyro information :(

**Comment by mifi** [comment]
did you try to enalbe more tracks? and try a different output format like matriska. someone made it work here #1465

**Reply by blurred82** [reply]
I see only two tracks in my file (video + audio).
Now I think that it cant possible if the all gyro data is stored at the beginning or at the end of the file.
Then if you cut the file you lose this metadata,

**Reply by blurred82** [reply]
> did you try to enalbe more tracks?
![Screenshot_4](https://user-images.githubusercontent.com/124368049/237060205-2f7a931a-f9f8-46cf-b1ca-7c22db96c844.jpg)


How to enable more tracks?

**Reply by mifi** [reply]
ok, probably insta360 stores gyro data differently then. is this mp4 file the only file from your camera? maybe gyro is stored separately?

**Reply by mifi** [reply]
related: #1121

**Reply by blurred82** [reply]
> ok, probably insta360 stores gyro data differently then. is this mp4 file the only file from your camera? maybe gyro is stored separately?

Gyro data is stored for sure in .mp4 file from camera.
I copy just this one file from camera and when I put it to Gyroflow or Insta360 Studio i get stabilized video.
But how it is stored with movie files I dont know.
Here you have a short sample ;) https://we.tl/t-RvY5ePCerl
It would be great to lossless-cut these files 😍

**Reply by mifi** [reply]
ah ok! what you could try is to enable "preserve all MOV metadata" option and see if it makes a difference. another thing to try is to use [exiftool to copy all the data](https://exiftool.org/forum/index.php?topic=3440.0) from input to output file. not optimal but there hav ebeen discussions about including exiftool into losslesscut in the future.

**Comment by repusman** [comment]
Has anyone figured this out yet.  i have searched endlessly for a way to cut files and preserve gyrodata but haven't found anything.

**Comment by alex-plekhanov** [comment]
@blurred82, @repusman, Insta360 stores gyro data as a file footer (not inside mp4 atoms), so, traditional mp4 files cut software is not suitable here. I've created tool to cut my Insta360 OneR camera files, perhaps Insta360 Go2 has the same file format and tool can be used to process this camera files as well. You can try [it](https://github.com/alex-plekhanov/insvtools).

**Reply by blurred82** [reply]
Thanks,
i will check it.
Fortunetelly insta360 studio 2023 already can cut go2 files preserving gyro data 😀
![Screenshot_1 insta360](https://github.com/mifi/lossless-cut/assets/124368049/78a9d3e8-a097-48fc-8b41-007ee9c7970a)
